core/src/main/csharp/ch/cyberduck/core/CredentialManagerPasswordStore.cs

```diff
@@ -119,21 +120,29 @@ public override OAuthTokens findOAuthTokens(Host bookmark)
                 logger.info(string.Format("Fetching OAuth tokens from keychain for {0}", bookmark));
             }
             var target = ToUri(bookmark);
-            var cred = WinCredentialManager.GetCredentials(target.AbsoluteUri);
-            if (cred.Attributes is Dictionary<string, string> attrs
-                && attrs.TryGetValue("OAuth Access Token", out var accessToken))
+            foreach(Uri descriptor in target)
             {
-                attrs.TryGetValue("OAuth Refresh Token", out var refreshToken);
-                attrs.TryGetValue("OIDC Id Token", out var idToken);
-                long expiry = default;
-                if (attrs.TryGetValue("OAuth Expiry", out var expiryValue))
+                var cred = WinCredentialManager.GetCredentials(descriptor.AbsoluteUri);
+                if (cred.Attributes is Dictionary<string, string> attrs)
                 {
-                    long.TryParse(expiryValue, out expiry);
+                    attrs.TryGetValue("OAuth Access Token", out var accessToken);
+                    attrs.TryGetValue("OAuth Refresh Token", out var refreshToken);
+                    attrs.TryGetValue("OIDC Id Token", out var idToken);
+                    long expiry = default;
+                    if (attrs.TryGetValue("OAuth Expiry", out var expiryValue))
+                    {
+                        long.TryParse(expiryValue, out expiry);
+                    }
+                    OAuthTokens tokens = new(accessToken, refreshToken, new(expiry), idToken);
+                    if(tokens.validate())
+                    {
+                        return tokens;
+                    }
+                    // Continue
```

Comment:
```suggestion
                    // continue: fallback to old behaviour for backwards compatibility
```

Reply:
One nitpick here:
Yes, fall back, but shouldn't _all_ OAuth Uris be tried?
This checks the first, and then returns to base-implementation - which should regardless.

Reply:
Good catch. Should only fallback to base implementation after trying all descriptors.

Reply:
Slipped through the first time looking at it.